src/core/algorithms/ind/ind_algorithm.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <list>
+#include <string>
+#include <unordered_map>
+
+#include "algorithms/algorithm.h"
+#include "ind.h"
+#include "tabular_data/input_tableset_type.h"
+
+namespace algos {
+
+class INDAlgorithm : public Algorithm {
+protected:
+    config::InputTableSet input_table_set_;
+
+public:
+    struct DatasetInfo {
+        std::string table_name;
+        std::vector<std::string> header;
+    };
+    using DatasetsOrder = std::vector<DatasetInfo>;
+    using IND = model::IND;
+    using INDList = std::list<IND>;
+
+public:
+    explicit INDAlgorithm(std::vector<std::string_view> phase_names);
+
+    virtual DatasetsOrder const& GetDatasetsOrder() const = 0;
```

Comment:
I don’t really like the fact that every developer who will implement algorithms for finding inclusion dependencies will need to implement this function.
Firstly, the user himself knows the datasets in what order he specified (this will be true after we switch to using a boost multitoken and manually tasking the list of tables) --- therefore, he must also know the order in which the datasets are specified.
Secondly, if we still want to give this information, then it is better to do it in a uniform way for all algorithms (which will be inherited from this class).
For example, we could simply add the following method:
```cpp
config::InputTableSet const& GetInputTableSet() const {
    return input_table_set_;
}
```
This will work well as we can get any information we need (table name/header).

Reply:
Also, in order to avoid the need to provide the user information about the order of the specified tables with the corresponding column names, we can add a new user-friendly `RawIND` class, which will contain information about the names and indices of the table and the names and indices of the columns themselves.
```cpp
namespase model {

struct RawIND {
   struct AttributeInfo {
       unsigned int attribute_id;
       string attribute_name;
       unsigned int column_id;
       string column_name;
   }; 
   using INDSide = std:vector<AttributeInfo>;
   
   INDSide lhs, rhs;
};

};
```
And then add to existing `INDAlgorithm` class methods to transform dependencies from `model::IND` to `model::RawIND`.
For example we can add public method that takes one dependency and gives a `RawIND`: `INDAlgorithm::ToRawIND(model::IND const& ind)`
And then add public method `INDAlgorithm::GetRawINDs()`.

But if we want to do this, then it is better to think about naming: I would prefer to rename the existing class `model::IND` to `model::INDCC`, and suggested `model::RawIND` to `model::IND`

Reply:
It seems to me that we don’t need this `DatasetInfo` at all. We have `input_table_collection_,` from which we can get both the name of the relation and the names of the columns.